### ISSUE TITLE
PopupWindow smart positioner/anchor API

### ISSUE THREAD

**Issue by jakobhellermann** [slint-ui/slint#4870]
When a PopupWindow is about to be shown in way that it would be cut by the end of the screen (or the window, for inline popup), we should actually place that popup in a proper place.

Wayland uses a concept of positioner: https://wayland.app/protocols/xdg-shell#xdg_positioner

the PopupWindow should have an anchor rectangle and so on so that the platform can petition the popup properly. 

**Update (2026):** this was kind of fixed long ago by moving the popup in internal/core/window/popup.rs, but it is not as flexible as what the xdg_positioner API have (anchors, gravity, ...) so we need to have properties in our PopupWindow to constraint the popup location, and implementation for the native popup and emulated popup. 

### Original message

basically https://github.com/slint-ui/slint/issues/3345 for the case where the `ComboBox` is at the bottom of the screen, which wasn't fixed yet as acknowledged in https://github.com/slint-ui/slint/pull/3861.

![image](https://github.com/slint-ui/slint/assets/22177966/8cd89ff6-cc75-4d7c-ad4c-12c62c9afdb0)

**Comment by ogoffart** [comment]
Thanks for filling an issue.

The first bug is that the popup is being clipped by the window. This is tracked in https://github.com/slint-ui/slint/issues/2375

Even if that problem was fixed, we would still have a problem if we were at the bottom of the screen, so we'd need a way to move the popup depending on its position on the screen.

**Comment by Enyium** [comment]
> Even if that problem was fixed, we would still have a problem if we were at the bottom of the screen, so we'd need a way to move the popup depending on its position on the screen.

Note that, for cases like this, Windows [provides system settings](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-trackpopupmenu#remarks). Although this *may* not apply to combo boxes specifically.

**Comment by ogoffart** [comment]
Closing as duplicate of https://github.com/slint-ui/slint/issues/6000#issuecomment-3053083364

**Comment by ogoffart** [comment]
This was not fixed after all.  I updated the title and description.
